packages/devtools/lib/src/memory/memory.dart

```diff
@@ -321,26 +332,129 @@ class MemoryScreen extends Screen with SetStateMixin {
           '${instanceRows.length} Instances of ${row.classRef.name}',
           (InstanceSummary row) => row.objectRef));
 
       table.setRows(instanceRows);
     } catch (e) {
-      framework.toast('Problem fetching instances $e', title: 'Error');
+      framework.toast(
+        'Problem fetching instances of ${row.classRef.name} $e',
+        title: 'Error',
+      );
     }
 
     table.onSelect.listen((InstanceSummary row) async {
       ga.select(ga.memory, ga.inspectInstance);
-      // TODO(terry): Handle clicking on an instance bring up variable inspector
+
+      try {
+        final Instance instance =
+            await memoryController.getObject(row.objectRef);
+
+        if (tableContainer.element.children.length > 2) {
+          tableContainer.element.children.removeLast();
+        }
+
+        tableContainer.add(_createInstanceView(
+          row.objectRef,
+          row.className,
+          instance.fields,
+        ));
+
+        tableContainer.element.scrollTo(<String, dynamic>{
+          'left': tableContainer.element.scrollWidth,
+          'top': 0,
+          'behavior': 'smooth',
+        });
+
+        // Allow inspection of the memory object.
+        memoryDataView.showFields(instance.fields);
+      } catch (e) {
+        framework.toast(
+          'Problem fetching instance ${row.objectRef} $e',
+          title: 'Error',
+        );
+      }
     });
 
     return table;
   }
 
+  CoreElement _createInstanceView(
+    String objectRef,
+    String className,
+    List<BoundField> fields,
+  ) {
+    final MemoryDescriber describer = (BoundField field) async {
+      if (field == null) {
+        return null;
+      }
+
+      final dynamic value = field.value;
+
+      if (value is Sentinel) {
+        return value.valueAsString;
+      }
+
+      if (value is TypeArgumentsRef) {
+        return value.name;
+      }
+
+      final Instance ref = value;
+
+      if (ref.valueAsString != null && !ref.valueAsStringIsTruncated) {
+        return ref.valueAsString;
+      } else {
+        final dynamic result = await serviceManager.service.invoke(
+          debuggerState.isolateRef.id,
+          ref.id,
+          'toString',
+          <String>[],
+          disableBreakpoints: true,
+        );
+        if (result is ErrorRef) {
+          return '${result.kind} ${result.message}';
+        } else if (result is InstanceRef) {
+          final String str = await _retrieveFullStringValue(result);
+          return str;
+        }
+      }
+    };
+
+    memoryDataView = MemoryDataView(memoryController, describer);
+
+    return div(
+        c: 'table-border table-virtual memory-table margin-left debugger-menu')
+      ..layoutVertical()
+      ..add(<CoreElement>[
+        div(
+          text: '$objectRef of $className',
+          c: 'memory-inspector',
+        ),
+        memoryDataView.element,
+      ]);
+  }
+
+  Future<String> _retrieveFullStringValue(InstanceRef stringRef) async {
```

Comment:
this seems like a helper to place at some place common.

Reply:
Added a TODO to consolidate to one for debugger and memory.